processing/src/main/java/io/druid/query/QueryToolChest.java

```diff
@@ -186,13 +205,83 @@
    * be queried.  It can use whatever criteria it wants in order to do the pruning, it just needs to
    * return the list of Segments it actually wants to see queried.
    *
-   * @param query The query being processed
+   * @param query    The query being processed
    * @param segments The list of candidate segments to be queried
-   * @param <T> A Generic parameter because Java is cool
+   * @param <T>      A Generic parameter because Java is cool
+   *
    * @return The list of segments to actually query
    */
   public <T extends LogicalSegment> List<T> filterSegments(QueryType query, List<T> segments)
   {
     return segments;
   }
+
+  /**
+   * @param filter Filter input to optimize
+   *
+   * @return Returns an optimized {@link DimFilter}
+   * Optimization is done by rewriting all ExtractionDimFilter of type {@link LookupExtractionFn} as a clause of literals {@link SelectorDimFilter}
+   */
+  public static DimFilter optimizeExtractionDimFilter(final DimFilter filter)
```

Comment:
I just noticed this is a method on the ToolChest.  I'm fairly certain it's just there to be able to share code.  Can we move it into a `static` method on something...

Errr, it is a `static` method, can we move it to the top of the class.  Generally, the code in Druid has static fields first, then static methods then constructor then class methods then private methods.  It would be nice to maintain that general flow to the code.